src/main/java/io/leonis/subra/game/rule/DirectFreeKickRule.java

```diff
@@ -0,0 +1,31 @@
+package io.leonis.subra.game.rule;
+
+import java.util.*;
+import io.leonis.zosma.game.Rule;
+import io.leonis.subra.game.data.*;
+import io.leonis.subra.game.data.Referee.Command;
+
+/**
+ * The Class DirectFreeKickRule.
+ *
+ * @author Rimon Oz
+ */
+public class DirectFreeKickRule implements Rule<Referee.Supplier, TeamColor> {
+
+  @Override
+  public boolean test(final Referee.Supplier refereeSupplier) {
+    return !this.getViolators(refereeSupplier).isEmpty();
+  }
+
+  @Override
+  public Set<TeamColor> getViolators(final Referee.Supplier refereeSupplier) {
```

Comment:
Why a set?

Reply:
see `Rule` in `zosma`, if it needs to be changed then we have to change it there

Reply:
i missed it somehow. seems reasonable